src/cpu.c

```diff
@@ -666,10 +677,42 @@ static int cpu_read(void) {
       cpu_stage(cpu, COLLECTD_CPU_STATE_SOFTIRQ, (derive_t)atoll(fields[7]),
                 now);
 
-      if (numfields >= 9)
+      if (numfields >= 9) {
         cpu_stage(cpu, COLLECTD_CPU_STATE_STEAL, (derive_t)atoll(fields[8]),
                   now);
+
+        if (numfields >= 10) { /* Guest (since Linux 2.6.24) */
```

Comment:
Please close the `>= 9` block here and out-dent the `>= 10` and `>= 11` cases. I.e.:

```c
if (numfields >= 9) {
  /* … */
}
if (numfields >= 10) {
  /* … */
}
if (numfields >= 11) {
  /* … */
}
```

Reply:
Should I get the (numfields >= 9) block out too? It is part of the (numfields >= 8) block, and that's why I chose to nest (numfields >= 10) and (numfields >= 11).

Reply:
Yes, that'd be great :)

Reply:
Done in 092b2d8. If needed, I'll rebase/squash everything after the discussions.